pkg/3scale/amp/component/s3.go

```diff
@@ -27,143 +23,9 @@ func NewS3(options *S3Options) *S3 {
-					}
-				}
-			}
-		}
-	}
-}
-
-// Remove the RWX_STORAGE_CLASS parameter because it is used only for the system-storage PersistentVolumeClaim
-func (s3 *S3) RemoveRWXStorageClassParameter(template *templatev1.Template) {
-	for paramIdx, param := range template.Parameters {
-		if param.Name == "RWX_STORAGE_CLASS" {
-			template.Parameters = append(template.Parameters[:paramIdx], template.Parameters[paramIdx+1:]...)
-			break
-		}
-	}
-}
-
-func (s3 *S3) getNewCfgMapElements() []v1.EnvVar {
-	return []v1.EnvVar{
-		envVarFromConfigMap("FILE_UPLOAD_STORAGE", "system-environment", "FILE_UPLOAD_STORAGE"),
-		envVarFromSecret(AwsAccessKeyID, s3.Options.AwsCredentialsSecret, AwsAccessKeyID),
-		envVarFromSecret(AwsSecretAccessKey, s3.Options.AwsCredentialsSecret, AwsSecretAccessKey),
-		envVarFromSecret(AwsBucket, s3.Options.AwsCredentialsSecret, AwsBucket),
-		envVarFromSecret(AwsRegion, s3.Options.AwsCredentialsSecret, AwsRegion),
-		envVarFromSecretOptional(AwsProtocol, s3.Options.AwsCredentialsSecret, AwsProtocol),
-		envVarFromSecretOptional(AwsHostname, s3.Options.AwsCredentialsSecret, AwsHostname),
-		envVarFromSecretOptional(AwsPathStyle, s3.Options.AwsCredentialsSecret, AwsPathStyle),
-	}
-}
-
-func (s3 *S3) AddCfgMapElemsToSystemBaseEnv(objects []common.KubernetesObject) {
-	newCfgMapElements := s3.getNewCfgMapElements()
-	for _, obj := range objects {
-		dc, ok := obj.(*appsv1.DeploymentConfig)
-		if ok {
-			if dc.ObjectMeta.Name == "system-app" || dc.ObjectMeta.Name == "system-sidekiq" {
-				if dc.ObjectMeta.Name == "system-app" {
-					dc.Spec.Strategy.RollingParams.Pre.ExecNewPod.Env = append(dc.Spec.Strategy.RollingParams.Pre.ExecNewPod.Env, newCfgMapElements...)
-				}
-
-				for containerIdx := range dc.Spec.Template.Spec.Containers {
-					container := &dc.Spec.Template.Spec.Containers[containerIdx]
-					container.Env = append(container.Env, newCfgMapElements...)
-				}
-			}
-		}
-	}
-}
-
-func (s3 *S3) AddS3PostprocessOptionsToSystemEnvironmentCfgMap(objects []common.KubernetesObject) {
-	var systemEnvCfgMap *v1.ConfigMap
-
-	for objIdx := range objects {
-		obj := objects[objIdx]
-		cfgmap, ok := obj.(*v1.ConfigMap)
-		if ok {
-			if cfgmap.Name == "system-environment" {
-				systemEnvCfgMap = cfgmap
-				break
-			}
-		}
-	}
-
-	systemEnvCfgMap.Data["FILE_UPLOAD_STORAGE"] = "s3"
-}
-
-func (s3 *S3) RemoveSystemStoragePVC(objects []common.KubernetesObject) []common.KubernetesObject {
-	res := objects
-
-	for idx, obj := range res {
-		pvc, ok := obj.(*v1.PersistentVolumeClaim)
-		if ok {
-			if pvc.ObjectMeta.Name == "system-storage" {
-				res = append(res[:idx], res[idx+1:]...) // This deletes the element in the array
-				break
-			}
-		}
-	}
-
-	return res
-}
-
+// TODO Template-only object. Decide what to do with this component.
```

Comment:
why don't you move it to templates package?

Reply:
I thought about it but I left it for a future PR where we might decide to remove it completely. That's why I left the TODO there.
We still need some parts of it in the component package anyway (constants) so I preferred to leave it this way at the moment (same applies with the s3_options file).